File: Dev/C++/Mascaret/src/VEHA/Entity/PathSpecification.cpp
#include "VEHA/Entity/PathSpecification.h"
#include "VEHA/Entity/PointSpecification.h"
#include "VEHA/Kernel/PrimitiveType.h"
#include "VEHA/Kernel/Model.h"
#include "Tools/utils.h"

namespace VEHA
{

PathSpecification::PathSpecification()
:TopologicalSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("path"))
{
}
PathSpecification::PathSpecification(const string& str)
:TopologicalSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("path"))
{
//TODO: parse string
}
PathSpecification::~PathSpecification()
{
}

shared_ptr<ValueSpecification> PathSpecification::clone() const
{
	throw std::runtime_error("can't clone a PathSpecification");
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Entity/PathSpecification.cs
using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class PathSpecification : TopologicalSpecification
    {

        private List<PointSpecification> points = new List<PointSpecification>();
        public List<PointSpecification> Points
        {
            get { return points; }
            set { points = value; }
        }

        public PathSpecification()
            : base(MascaretApplication.Instance.Model.getBasicType("path"))
        {
        }

        public PathSpecification(string str)
            : base(MascaretApplication.Instance.Model.getBasicType("path"))
        {
            //TODO parse string
        }

        public override ValueSpecification clone()
        {
            throw new NotImplementedException();
        }
    }
}


